cpp/src/arrow/array-test.cc

```diff
@@ -456,12 +484,32 @@ TYPED_TEST(TestPrimitiveBuilder, TestAppendNull) {
     ASSERT_OK(this->builder_->AppendNull());
   }
 
-  std::shared_ptr<Array> result;
-  ASSERT_OK(this->builder_->Finish(&result));
+  std::shared_ptr<Array> out;
+  ASSERT_OK(this->builder_->Finish(&out));
+  auto result = std::dynamic_pointer_cast<typename TypeParam::ArrayType>(out);
 
   for (int64_t i = 0; i < size; ++i) {
     ASSERT_TRUE(result->IsNull(i)) << i;
   }
+
+  TestInitialized(*result);
```

Comment:
Sometimes we only call `TestInitialized`, sometimes we call `IsZeroPadded`, sometimes both... is there a rationale? Otherwise I'd rather call them both always.

Reply:
You're right, I've changed it in the commits I've just pushed.

Reply:
Oh, at this exact place it's actually intended: for primitive / adaptive int builders we have separate tests for uninitialized memory and for zero-padding. For more complex builders we check it both at places like `ValidateBasicListArray`.